client/app/telecom/pack/xdsl/modem/service/pack-xdsl-modem-service.controller.js

```diff
@@ -0,0 +1,359 @@
+angular.module('managerApp').controller('XdslModemServiceCtrl', class XdslModemServiceCtrl {
+  /* @ngInject */
+  constructor($q, $stateParams, $translate, $uibModal, OvhApiXdsl, TucPackXdslModemMediator,
+    TucToast) {
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.$q = $q;
+    this.OvhApiXdsl = OvhApiXdsl;
+    this.TucToast = TucToast;
+    this.$uibModal = $uibModal;
+    this.mediator = TucPackXdslModemMediator;
+
+    this.service = {};
+  }
+
+  getService() {
+    this.getBlocIP();
+    this.getCallWaiting();
+    this.getContentSharing();
+    this.getFTP();
+    this.getIpsecAlg();
+    this.getSipAlg();
+    this.getUPnP();
+  }
+
+  getBlocIP() {
+    return this.OvhApiXdsl.Modem().BlocIp().v6().get({
+      xdslId: this.$stateParams.serviceName,
+    }).$promise.then(({ data }) => {
+      if (data === 'enabled') {
+        this.service.blocIP = true;
+      } else {
+        this.service.blocIP = false;
+      }
+      this.service.blocIPTmp = this.service.blocIP;
+      this.service.isBlocIP = true;
+      return data;
+    }).catch(() => {
+      this.service.isBlocIP = false;
+    });
+  }
+
+  changeBlocIp(value) {
+    if (value) {
+      this.updateBlocIp('enabled');
+    } else {
+      this.updateBlocIp('disabled');
+    }
+  }
+
+  updateBlocIp(status) {
+    return this.OvhApiXdsl.Modem().BlocIp().v6().post({
+      xdslId: this.$stateParams.serviceName,
+    },
+    {
+      status,
+    }).$promise.then((data) => {
```

Comment:
According to the API response class:

```suggestion
    }).$promise.then((xdslTask) => {
```